plugins/CorePluginsAdmin/Controller.php

```diff
@@ -87,6 +99,13 @@ public function __construct(Translator $translator, Plugin\SettingsProvider $set
 
     public function uploadPlugin()
     {
+        if (!$this->passwordVerify->isPasswordCorrect(
+            Piwik::getCurrentUserLogin(),
+            Common::getRequestVar('confirmPassword')
+        )) {
+            throw new \Exception($this->translator->translate('Login_LoginPasswordNotCorrect'));
+        }
+
         static::dieIfPluginsAdminIsDisabled();
         Piwik::checkUserHasSuperUserAccess();
```

Comment:
we need to make sure to first perform several checks before performing the password check. Meaning we need to make sure
* admin is not disabled
* user has super user access. 
* Plugin upload is enabled
* Nonce check is correct

Otherwise could be potentially misused for brute forcing password etc. (it would actually not really be possible) but it's a lot safer to do the password check only after a nonce etc. Could even check it only shortly before `$this->configureView('@CorePluginsAdmin/uploadPlugin');` so we make sure the basic things are correct.

Works otherwise, we only need to make sure to move the check further down

Reply:
Fair enough, have moved it to just before validation of the uploaded file